rocAL/rocAL/source/loaders/audio/audio_loader_sharded.cpp

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright (c) 2019 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "audio_loader_sharded.h"
+
+AudioLoaderSharded::AudioLoaderSharded(void* dev_resources):
+        _dev_resources(dev_resources) {
+    _loader_idx = 0;
+}
+
+void AudioLoaderSharded::set_prefetch_queue_depth(size_t prefetch_queue_depth) {
+    if(prefetch_queue_depth <= 0)
+        THROW("Prefetch queue depth value cannot be zero or negative");
+    _prefetch_queue_depth = prefetch_queue_depth;
+}
+
+std::vector<std::string> AudioLoaderSharded::get_id() {
+    if(!_initialized)
+        THROW("get_id() should be called after initialize() function");
+    return _loaders[_loader_idx]->get_id();
+}
+
+decoded_image_info AudioLoaderSharded::get_decode_image_info() {
+    return _loaders[_loader_idx]->get_decode_image_info();
+}
+
+crop_image_info AudioLoaderSharded::get_crop_image_info() {
```

Comment:
Crop Image info is not required for audio right?
If it is not required it must throw error. Please check

Reply:
I am removing this funbction